src/logic/exceptions/registry.ts

```diff
@@ -0,0 +1,28 @@
+export interface ExceptionContent {
+  description: string
+  uiMessage?: string
+}
+
+/**
+ * When creating a new error type, please try to group them semantically
+ * with the existing errors in the same hundred. For example, if it's
+ * related to fetching data from the backend, add it to the 6xx errors.
+ * This is not a hard requirement, just a useful convention.
+ */
+const registry: Record<number, ExceptionContent> = {
+  0: {
+    description: 'No such error code',
+  },
+  100: {
+    description: 'Invalid input in the address field',
+    uiMessage: 'Please enter a valid address',
+  },
+  600: {
+    description: 'Error fetching token list',
+  },
+  601: {
```

Comment:
Wouldn't it be easier to use if we add a named relation? So we can have the numbers to index in the document but at code level , print the code number in the exception if necessary but using it as an enum type: ERROR_FETCHING_TOKEN_LIST, ERROR_FETCHING_TOKEN_BALANCES....

Reply:
I also thought about that. It would make the autocomplete nicer and type-checking more powerful.

An enum per se won't work here because they can only be strings or numbers, and we have records here. But a simple object will work.

The only problem is see is that you'd have to come up with a name every time, and the name will be a bit redundant together with the description.

So maybe we could stick with numeric keys like so?
```
{
  e601: {
    description: 'Error fetching balances',
  }
}
```

And then you could use it like so:
```
logError(ErrorCodes.e601)
```

wdyt?

Reply:
It depends on how fast the error list grows. For me the way that this came to my mind is that I found this code:

```
catch (e) {
  logError(601, e.message, false)
  return
}
```

And I had to open the exceptions registry in order to know what we were logging there

Reply:
I've ditched the `uiMessage` to make the registry a simple string enum. So now you can just jump to definition.
Do you think that would work?

Reply:
Something around https://tsplay.dev/mL9YAW won't work?

it requires to have two places to update (the enum and the registry), but you won't be able to add a new registry entry if it's not present in the enum.

Reply:
If we really want to have human-readable names for errors, it could look like this:

```
enum ErrorCodes {
  NO_SUCH_ERROR_CODE = '0: No such error code',
  ADDRESS_FIELD_INVALID_INPUT = '100: Invalid input in the address field',
}
```

I personally think it adds mental overhead (coming up with short names which are also not broken English) and redundancy (name repeats the description).

However, I agree it certainly makes it more convenient to read the code.

Reply:
Yes, that last proposal would make sense adding the error code at the start of the property name. But I'm fine with the last approach suggested as most of code editor should show the message when hovering the propoerty